chore: refactored some of the logic to restore burner wallet usage wh…

davedumto opened the pull request:
…ile maintaining the usage of cartridge.

# Pull Request | Checkers
## 1. Restore Burner Wallet Functionality
- Closes #38

## 2. Brief Description
The current implementation lacks burner wallet functionality which limits wallet management options. This PR restores burner wallet support while maintaining cartridge integration, giving users flexibility in how they interact with the game.

## 3. Changes Made
- [x] Re-implemented burner wallet creation and management using @dojoengine/create-burner
- [x] Modified main.tsx to initialize burner manager with setupBurnerManager
- [x] Updated DojoContext to properly handle burner wallet state
- [x] Added CreateBurner component in the initgame file
- [x] Integrated burner functionality while preserving cartridge support
- [x] Added wallet switching and clearing capabilities

## 4. Checklist
- [x] Tested burner wallet creation and management
- [x] Verified compatibility with existing cartridge implementation
- [x] Code follows project standards
- [x] Functionality tested in local environment

## 5. Evidence (Optional)
![Screenshot 2025-01-03 at 10 00 28 AM](https://github.com/user-attachments/assets/e82180f2-8a3f-4a23-9d9e-2c82ab94f34e)

davedumto committed:
chore: refactored some of the logic to restore burner wallet usage while maintaining the usage of cartridge.

Nadai2010 reviewed (changes requested):
Hi @davedumto good job, i think that you need to add CreateBurner for Create Game, otherwise you won't be able to play, the idea is that by resetting the burner wallet you can play with these accounts.

![imagen](https://github.com/user-attachments/assets/c3efb316-3447-4e98-85bd-02c654b68745)

davedumto commented:
okay let me add that to the create game logic.

davedumto committed:
fix: made the logic to be able to use both burner wallets and cartridge to create game and join game

davedumto committed:
little change

davedumto committed:
fix: made one last fix to the checkers logic

Nadai2010 commented on client/src/components/JoinRoom.tsx:
```diff
@@ -91,6 +107,7 @@ function JoinRoom({ }: { sdk: SDK<typeof schema> }) {
           zIndex: 2,
         }}>
         <ControllerButton />
+        <CreateBurner />
```
Add component the `CreateBurner` first and then `ControllerButton` so all game tabs keep the same structure

davedumto replied:
Okay I am doing that right now

Nadai2010 reviewed (changes requested):
Everything works perfectly, both the Controller and the Create Burner, you connect the accounts and you can create rooms and play. Good job

davedumto committed:
chore: changed the position of the create burner component

gianalarcon commented:
Awesome, please @davedumto provide a quick loom video of the full game-play pls, similar as this video [Video](https://www.loom.com/share/9e0e9452c54a4a4b932875db53ff4446)

davedumto commented:
> Awesome, please @davedumto provide a quick loom video of the full game-play pls, similar as this video [Video](https://www.loom.com/share/9e0e9452c54a4a4b932875db53ff4446)

will do that right now

davedumto commented:
> Awesome, please @davedumto provide a quick loom video of the full game-play pls, similar as this video [Video](https://www.loom.com/share/9e0e9452c54a4a4b932875db53ff4446)

This is the [Video](https://www.loom.com/share/8edff884d2a04a309aabc29969ddfc30?sid=b2d8819e-6740-4818-ae27-1fa861231fce)

gianalarcon commented:
Looks good to me. Please just update the readme and a small section to mention about burner wallet with few instructions. After that we are ready to merge

davedumto committed:
Update README.md

I added a little guide on how to use the Burner Wallets to create and play game

davedumto commented:
@gianalarcon I have updated the readme. kindly look through it
